### PR TITLE
Facilitator readme file

### LINKED ISSUE #227: Update readme, changelog & publish npm
- Update readme
- Update changelog
- Update package.json version
- Publish facilitator as npm

### PR DESCRIPTION
**Pull request by sarvesh-ost**
Blocked by https://github.com/mosaicdao/mosaic-chains/pull/197

This PR adds detailed readme for facilitator. 

partially fixes #227

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,32 +1,214 @@
...
+
+```
+
+`<graph-rpc-admin-url>` and `<graph-ipfs-url>` was displayed while starting origin and auxiliary chain.
+
+By default subgraph command will deploy subgraph for OST gateways, optionally it also accepts `--mosaic-config`, `--gateway-config` and `--gateway-address` option to deploy subgraph for other gateways. 
+
+Mosaic config i.e `mosaic.json` can be found in [here](https://github.com/mosaicdao/mosaic-chains/tree/develop/chains) inside `<origin-chain>` folder.
+
+Gateway config i.e. `<gatewayaddress>.json` can be found in [here](https://github.com/mosaicdao/mosaic-chains/tree/develop/chains) inside `<origin-chain>/<auxiliary-chain>/<gateway-address.json`.
```
Comment (abhayks1) [inline README.md]:
```suggestion
Gateway config i.e. `<gatewayaddress>.json` can be found [here](https://github.com/mosaicdao/mosaic-chains/tree/develop/chains) inside `<origin-chain>/<auxiliary-chain>/<gateway-address.json`.
```
Same comment as above.

File: README.md [hunk 2]
```diff
@@ -1,32 +1,214 @@
...
+
+More documentation about `subgraph` command can be found [here](https://github.com/mosaicdao/mosaic-chains#subgraph-deployment).
+
+**4. Facilitator init**: Facilitator init command initialize the facilitator which creates database, worker addresses and create facilitator config file. Facilitator config file is needed to start facilitator.
```
Comment (abhayks1) [inline README.md]:
```suggestion
**4. Facilitator init**: Facilitator init command initializes the facilitator. The command creates database, worker addresses and create facilitator config file. Facilitator config file is mandatory to start facilitator.  
```

File: README.md [hunk 3]
```diff
@@ -1,32 +1,214 @@
...
+
+**4. Facilitator init**: Facilitator init command initialize the facilitator which creates database, worker addresses and create facilitator config file. Facilitator config file is needed to start facilitator.  
+
+```
+facilitator init --mosaic-config <mosaic-config> --aux-chain-id <aux-chain-id> --origin-password <origin-password> --auxiliary-password <auxiliary-password> --origin-rpc <origin-rpc> --auxiliary-rpc <auxiliary-rpc> --origin-graph-ws <origin-graph-ws> --origin-graph-rpc <origin-graph-rpc> --auxiliary-graph-ws <auxiliary-graph-ws> --auxiliary-graph-rpc <auxiliary-graph-rpc> --db-path <db-path> --force
+
 ```
 
-* Replace `<gateway-config>` with location where gateway config is present.
 * Replace `<mosaic-config>` with location where mosaic config is present.
```
Comment (abhayks1) [inline README.md]:
```suggestion
* Replace `<mosaic-config>` with file location where mosaic config is present.
```

File: README.md [hunk 4]
```diff
@@ -1,32 +1,214 @@
...
+* Replace `<auxiliary-graph-ws>` with auxiliary ws graph endpoint.
+* Replace `<auxiliary-graph-rpc>` with auxiliary rpc graph admin endpoint.
 * Replace `<db-path>` with the database path. It is the path for `sqlite` database. If not provided,it would create it.
 * `--force` option is used to forcefully override facilitator config. It is optional parameter.
-	
-## Facilitator Start
 
-Facilitator start command will start the facilitator. 
+
+Facilitator init can be also done with `--gateway-config <gateway-config>` option. Gateway config i.e. `<gatewayaddress>.json` can be found in [here](https://github.com/mosaicdao/mosaic-chains/tree/develop/chains) inside `<origin-chain>/<auxiliary-chain>/<gateway-address.json`.
```
Comment (abhayks1) [inline README.md]:
```suggestion
Facilitator init can also be done with `--gateway-config <gateway-config>` option. Gateway config i.e. `<gatewayaddress>.json` can be found [here](https://github.com/mosaicdao/mosaic-chains/tree/develop/chains). It's inside `<origin-chain>/<auxiliary-chain>/<gateway-address.json`.
```
Same comment as above.

File: README.md [hunk 5]
```diff
@@ -1,32 +1,214 @@
...
+
+Replace <gateway-config> with location where gateway config is present.
```
Comment (abhayks1) [inline README.md]:
```suggestion
Replace <gateway-config> with file location where gateway config is present.
```

File: README.md [hunk 6]
```diff
@@ -1,32 +1,214 @@
...
+export MOSAIC_ADDRESS_PASSW_<Address>=<origin-password>
+export MOSAIC_ADDRESS_PASSW_<Address>=<auxiliary-password>
+```
+Above variables will also be produced with the output of `facilitator init` command.
+
+**6. Fund facilitator workers for gas and bounty**: Workers created in facilitator init step needs to be funded for gas and bounty on both chains.
+
+   *Gas on origin*: Facilitator needs to pay for gas on origin chain. Origin worker address created in `facilitator init` step must be funded to pay for transaction fee. 
+   
+   There are various ways to fund on origin chain, for testnet any public faucet can be used. Below web3 transaction can also be done, if there is fund in an existing key.
```
Comment (abhayks1) [inline README.md]:
```suggestion
   There are various ways to fund worker on origin chain. For testnet any public faucet can be used. Below web3 transaction can be done, if there is fund in an existing key.
```

File: README.md [hunk 7]
```diff
@@ -1,32 +1,214 @@
...
+   
+      
+      web3.eth.sendTransaction(
+      {
+         from:<funder_address>, 
+          to: <origin_worker>,
+          value:<Fund_in_wei
+      })
+      
+   *Gas on auxiliary*: Facilitator also needs to pay for transaction fee on auxiliary chain. For testnet [mosaic faucet](https://github.com/mosaicdao/faucet) can be used to get fund. Alternatively, OST on value chain can also be converted to base token on auxiliary chain in order to pay for gas. This is explained in detail in section [request stake]().
```
Comment (abhayks1) [inline README.md]: [request stake]() can be hyperlinked.

File: README.md [hunk 8]
```diff
@@ -1,32 +1,214 @@
...
+   
+   *Value token for bounty*: Facilitator needs to stake bounty to gateway/co-gateway in order to perform stake & mint and redeem & unstake. Bounty token on the origin chain for existing mosaic chain is simple token and on the auxiliary chain bounty is base token. 
+   
+   On testnet, bounty can be funded to workers using [mosaic faucet](https://github.com/mosaicdao/faucet).
+   
+**7. White list workers**: origin and auxiliary workers should be whitelisted in stakepool and redeem pool contract respectively. Below commands will whitelist the workers.
```
Comment (abhayks1) [inline README.md]:
```suggestion
**7. White list workers**: origin and auxiliary workers should be whitelisted in stakepool and redeem pool contracts respectively. Below commands will whitelist the workers. 
```

File: README.md [hunk 9]
```diff
@@ -1,32 +1,214 @@
...
+   
+**7. White list workers**: origin and auxiliary workers should be whitelisted in stakepool and redeem pool contract respectively. Below commands will whitelist the workers. 
+
+   1. Clone git repository:
+   ```
+       git clone https://github.com/mosaicdao/mosaic-chains.git
+   ```
+   2. Set below enviornment variables
+   
+
```
Comment (abhayks1) [inline README.md]: why two new lines?

File: README.md [hunk 10]
```diff
@@ -1,32 +1,214 @@
...
+    export AUXILIARY_WORKER_EXPIRATION_HEIGHT='replace_with_auxiliary_expiration_height';
+
+origin and auxiliary worker addresses are generated with `facilitator init` step. 
+Mosaic config path for supported chain should be available on `~/.mosaic/<origin-chain>/mosaic.json` where `<origin-chain>` is origin chain identifier e.g. `ropsten`.  
+
+origin and auxiliary worker expiration height is block height from current block for which worker keys are whitelisted. If current block is 1000 and expiration height is set to 100 then worker keys will be whitelisted for 1100 block.
+
+   3. run command:
+     
+     npm run whitelist-workers
```
Comment (abhayks1) [inline README.md]: will he run this from facilitator repo?

Comment (sarvesh-ost) [reply README.md]: No, step 1 is about cloning mosaic-chains.

File: README.md [hunk 11]
```diff
@@ -1,32 +1,212 @@
...
+   
+   *Value token for bounty*: Facilitator needs to stake bounty to gateway/co-gateway in order to perform stake & mint and redeem & unstake. Bounty token on the origin chain for existing mosaic chain is simple token and on the auxiliary chain bounty is base token. 
+   
+   On testnet, bounty can be funded to workers using [mosaic faucet](https://github.com/mosaicdao/faucet).
+   
+**7. White list workers**: origin and auxiliary workers should be whitelisted in stakepool and redeem pool contracts respectively. Below commands will whitelist the workers. 
+
+   1. Clone git repository:
+   ```
+       git clone https://github.com/mosaicdao/mosaic-chains.git
```
Comment (puneet-khushwani-eth) [inline README.md]: we asked the user to install `mosaic-chains` globally above. Have them clone it again here might be a problem

File: README.md [hunk 12]
```diff
@@ -1,32 +1,212 @@
...
+Mosaic facilitator is an executable which enables atomic token transfers across two blockchains. Facilitator executable facilitates stake & mint and Redeem & unstake of any EIP20 token using mosaic gateways. 
 
-Facilitator init command will create facilitator config for an auxiliary chain. 
+## Prerequisite
+1. Docker version `18.09.1` or above.
+2. Docker compose version `1.23.2` or above.
+3. Node version `11` or above.
+4. Mosaic chains executable. 
+   ```
+   npm i @openst/mosaic-chains -g
```
[the remark this answers is not in the document]
Comment (sarvesh-ost) [reply README.md]: Simply `mosaic start ropsten`

[the remark this answers is not in the document]
Comment (sarvesh-ost) [reply README.md]: yes

File: README.md (hunk 11 above)
Comment (sarvesh-ost) [reply README.md]: It's for the whitelist-worker script. Any other idea?

File: README.md [hunk 13]
```diff
@@ -63,3 +244,7 @@ Facilitator can be started in below two ways :-
     * Replace `<gateway-config>` with the path to gateway-config.json.	    
     
   * **Note** : Both `--mosaic-config` and `--gateway-config` are together not allowed in command.
+
+
+    
+
```
Comment (puneet-khushwani-eth) [inline README.md]: We haven't added a section in here to help users to just move tokens around using facilitators hosted by others.

**Pull request merged**